java/custom/com/nokia/mid/s40/bg/BGUtils.java

```diff
@@ -5,48 +5,15 @@
 import com.sun.midp.midletsuite.MIDletSuiteStorage;
 
 public class BGUtils {
-    private static boolean launchMIDletCalled = false;
-
     private static native int getFGMIDletNumber();
     private static native String getFGMIDletClass();
-    static native void waitUserInteraction();
+    public static native void maybeWaitUserInteraction(String midletClassName);
 
-    /* Start the FG MIDlet when the page becomes visible and if
-       launchIEMIDlet hasn't been called (we want launchIEMIDlet
-       to launch the MIDlet if possible) */
     public static void setBGMIDletResident(boolean param) {
-        BGUtils.waitUserInteraction();
-        BGUtils.startMIDlet();
+        AmsUtil.executeWithArgs(MIDletSuiteStorage.getMIDletSuiteStorage(), 0, BGUtils.getFGMIDletNumber(),
```

Comment:
Nit: this could use a comment explaining why we're launching the foreground midlet here.